Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/AuthMethodManager.java

```diff
@@ -68,34 +64,36 @@ private AuthMethodManager(AuthMethodDetails authMethodDetails) {
         this.authMethodDetails = authMethodDetails;
         // initialize subscription manager when restore authentication
         if (this.authMethodDetails.getAuthMethod() != null) {
-            try {
-                getAzureManager().getSubscriptionManager().updateSubscriptionDetailsIfNull();
-            } catch (IOException e) {
-                // swallow exception
-            }
+            Objects.requireNonNull(getAzureManager()).getSubscriptionManager().updateSubscriptionDetailsIfNull();
         }
     }
 
     public static AuthMethodManager getInstance() {
         return LazyHolder.INSTANCE;
     }
 
-    public Azure getAzureClient(String sid) throws IOException {
-        if (getAzureManager() == null) {
-            throw new IOException(CANNOT_GET_AZURE_MANAGER);
-        }
-        Azure azure = getAzureManager().getAzure(sid);
-        if (azure == null) {
-            throw new IOException(String.format(CANNOT_GET_AZURE_BY_SID, sid));
+    @NotNull
+    public Azure getAzureClient(String sid) {
+        final AzureManager manager = getAzureManager();
+        if (manager != null) {
+            final Azure azure = manager.getAzure(sid);
+            if (azure != null) {
+                return azure;
+            }
         }
-        return azure;
+        final String error = "Failed to initialize request";
+        final String action = "Confirm you have already signed in with subscription: " + sid;
+        throw new AzureToolkitRuntimeException(error, action);
     }
 
-    public AppPlatformManager getAzureSpringCloudClient(String sid) throws IOException {
-        if (getAzureManager() == null) {
-            throw new IOException(CANNOT_GET_AZURE_MANAGER);
+    public AppPlatformManager getAzureSpringCloudClient(String sid) {
+        final AzureManager manager = getAzureManager();
+        if (manager != null) {
+            return getAzureManager().getAzureSpringCloudClient(sid);
         }
-        return getAzureManager().getAzureSpringCloudClient(sid);
+        final String error = "Failed to initialize request";
```

Comment:
duplicate with previous, create an exception `AzureToolkitNotSignedInException`?

Reply:
let's summarize such kind of common exceptions in the future.